packages/dds/tree/src/simple-tree/core/treeNodeKernel.ts

```diff
@@ -97,39 +114,19 @@ export class TreeNodeKernel implements Listenable<KernelEvents> {
 	 */
 	public generationNumber: number = 0;
 
-	#hydrated?: {
-		anchorNode: AnchorNode;
-		offAnchorNode: Set<Off>;
-	};
+	#hydrationState: HydrationState = () => {};
 
 	/**
 	 * Events registered before hydration.
 	 * @remarks
 	 * As an optimization these are allocated lazily as they are usually unused.
-	 */
-	#preHydrationEvents?: Listenable<KernelEvents> &
-		IEmitter<KernelEvents> &
-		HasListeners<KernelEvents>;
-
-	/**
-	 * Get the listener.
-	 * @remarks
-	 * If before hydration, allocates and uses `#preHydrationEvents`, otherwise the anchorNode.
-	 * This design avoids allocating `#preHydrationEvents` if unneeded.
-	 *
-	 * This design also avoids extra forwarding overhead for events from anchorNode and also
-	 * avoids registering for events that are unneeded.
+	 * The laziness also avoids extra forwarding overhead for events from anchorNode and also avoids registering for events that are unneeded.
```

Comment:
Should this say `anchor nodes` instead of `anchorNode`? I'm having a hard time parsing this.

Reply:
It's talking about the anchor node that belongs to this kernel I think, so I made that more clear.